### PR TITLE
ODS-5195 - Add MSSQL compose files for district specific and year specific 

### PR DESCRIPTION
**Pull request by bptillman**
In addition to creating the new templates, one thing to note is that `SQLSERVER_ADMIN_DATASOURCE` was added into all of the MSSQL templates and will need to be added to the .env file as well to properly test this. [This page](https://techdocs.ed-fi.org/display/EDFITOOLS/Docker+Deployment) was also updated to reflect this new addition.

No mustache templates were created for this since for MSSQL it is expected the databases will already be setup, compared to the postgres compose files that would be doing the db setup.

To test this you can do the following:
1. Add/update your .env file in the root of the repository to have the correct values for: `SQLSERVER_DATA_SOURCE`, `SQLSERVER_USER`, `SQLSERVER_PASSWORD`, and `SQLSERVER_ADMIN_DATASOURCE`. The admin one is a new one, so if you already had a .env file you will need to add this new one to it.
2. From the ODS-Implementation repo, run initdev for year specific so that the correct SQL databases get created. For example, you can run `initdev -InstallType YearSpecific -OdsTokens @("2019","2020")`
3. Edit your Windows host file found at C:\Windows\System32\drivers\etc\hosts to include mappings of your local network ip address to the year specific named sql datasources. Here is an example that maps a local network ip of 10.0.0.21 to a 2019 and 2020 ODS data source
```
10.0.0.21 EdFi_Ods_2019
10.0.0.21 EdFi_Ods_2020
```
4. From the `Compose\mssql` folder you can run the following command for testing the year specific file:
    1. `docker-compose --env-file ../../.env -f compose-year-specific-env-build.yml up --build`
5. In Postman, get the oauth token from `https://localhost/api/oauth/token`, and then attempt to post data to to one specific year instance, such as a student for 2019 by posting to this URL `https://localhost/api/data/v3/2019/ed-fi/students` with new student like below (and the token for auth)

```
{
    "studentUniqueId": "1234567",
    "birthDate": "2001-05-24",
    "firstName": "Ben",
    "lastSurname": "Tester",
    "personalTitlePrefix": "Mr."
}
```

6. You can then query the ODS database to confirm that the student now exists there.
7. Then you can repeat the same steps above, but for district specific
    1. So you would initdev like this: `initdev -InstallType DistrictSpecific -OdsTokens '255901;255902'`
    2. And docker compose like this: `docker-compose --env-file ../../.env -f compose-district-specific-env-build.yml up --build`

Two things to note when testing this:
1. Currently there is an [open ticket](https://tracker.ed-fi.org/browse/AA-1285) for AdminApp to work fully in docker for year and district specific, so if you are trying to use it as part of this testing, it will not work via docker but you can try to run it via Visual Studio instead.
2. The two additional files of `compose-district-specific-env.yml` and `compose-year-specific-env.yml` cannot be run yet since the updated images for those from this branch are not up on docker hub yet and do not get built until code is merged to master.

### CODE REVIEW

File: Compose/mssql/compose-district-specific-env-build.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+version: "3.8"
+
+services:
+  nginx:
+    build:
+      context: ../../Web-Gateway/Alpine
+      dockerfile: Dockerfile
+    environment:
+      ADMINAPP_VIRTUAL_NAME: "${ADMINAPP_VIRTUAL_NAME:-admin}"
```
Comment (vimayya) [inline Compose/mssql/compose-district-specific-env-build.yml:14]: Should be adminapp

File: Compose/mssql/compose-district-specific-env.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+version: "3.8"
+
+services:
+  nginx:
+      image: edfialliance/ods-api-web-gateway:${TAG}
+    environment:
+      ADMINAPP_VIRTUAL_NAME: "${ADMINAPP_VIRTUAL_NAME:-admin}"
```
Comment (vimayya) [inline Compose/mssql/compose-district-specific-env.yml:12]: Change virtual name to adminapp

File: .env.example [hunk 3]
```diff
@@ -20,6 +20,7 @@ ENCRYPTION_KEY=<base64-encoded 256-bit key>
 
 # For SQL Server only
 SQLSERVER_DATASOURCE=<DNS or IP Address of the SQL Server Instance, i.e. sql.somedns.org or 10.1.5.9,1433
```
Comment (vimayya) [inline .env.example:22]: Could SQLSERVER_DATASOURCE be changed to SQLSERVER_ODS_DATASOURCE for clarity

File: Compose/mssql/compose-year-specific-env-build.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+version: "3.8"
+
+services:
+  nginx:
+    build:
+      context: ../../Web-Gateway/Alpine
+      dockerfile: Dockerfile
+    environment:
+      ADMINAPP_VIRTUAL_NAME: "${ADMINAPP_VIRTUAL_NAME:-admin}"
```
Comment (vimayya) [inline Compose/mssql/compose-year-specific-env-build.yml:14]: Change virtual name to adminapp

Comment (vimayya) [reply Compose/mssql/compose-year-specific-env-build.yml:14]: I see it is wrong in compose-shared-instance-env-build.yml as well, could you update to adminapp there?

File: Compose/mssql/compose-district-specific-env-build.yml (hunk 1 above)
Comment (bptillman) [reply Compose/mssql/compose-district-specific-env-build.yml]: This has been updated.

File: Compose/mssql/compose-district-specific-env.yml (hunk 2 above)
Comment (bptillman) [reply Compose/mssql/compose-district-specific-env.yml]: This has been updated.

File: .env.example (hunk 3 above)
Comment (bptillman) [reply .env.example]: This has been updated.

File: Compose/mssql/compose-year-specific-env-build.yml (hunk 4 above)
Comment (bptillman) [reply Compose/mssql/compose-year-specific-env-build.yml]: This has been updated.

**Review by vimayya — APPROVED** [review]

**Pull request merged**